### PR TITLE
IDEMPIERE-5637 - Improve GridField to allow setting the default values of GridFieldVO programmatically with a method

### PR DESCRIPTION
**Pull request by d-ruiz**
…default value programmatically

https://idempiere.atlassian.net/browse/IDEMPIERE-5637

# Pull Request Checklist

- [x] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [x] My code follows the best practices of this project
- [x] I have performed a self-review of my own code
- [x] My code is easy to understand and review. 
- [x] I have checked my code and corrected any misspellings
- [ ] In hard-to-understand areas, I have commented my code.
- [x] My changes generate no new warnings
### Tests
- [x] I have tested the direct scenario that my code is solving
- [x] I checked all collaterals that can be affected by my changes, and tested other potential affected scenarios
- [x] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works
### Documentation
- [ ] I have made corresponding changes to the documentation as follows:
- - [ ] New feature (non-breaking change which adds functionality): I have created the New Feature page in the project wiki explaining the functionality and how to use it. If relevant, I have committed sample data to the core seed to have usable examples in GardenWorld.
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected): I have documented the change in a clear way that everyone in the community can understand the impact of the change.
- - [ ] Improvement (improves and existing functionality): This documentation is needed if the improvement changes the way the user interacts with the system or the outcome of a process/task changes. If it is just, for instance, a performance improvement, documentation might not be needed. 
- [ ] The changed/added documentation is in the project wiki (not privately-hosted pdf files or links pointing to a company website) and is complete and self-explanatory.

### CODE REVIEW

File: org.adempiere.base/src/org/compiere/model/GridField.java
Comment (d-ruiz) [inline org.adempiere.base/src/org/compiere/model/GridField.java:1]: If you guys I can move this to a new pull request. However, it is harmless and it makes the code much easier to read. Now that I am working on this class, it made sense to include this minor refactor here.

File: org.adempiere.base/src/org/compiere/model/GridField.java [hunk 1]
```diff
@@ -591,6 +599,13 @@ public void setInserting (boolean inserting)
 		m_inserting = inserting;
 	}   //  setInserting
 
+	public void setDefaultLogic(String defaultValue) {
+		m_vo.DefaultValue = defaultValue;
+	}
+
+	public void setDefault2Logic(String defaultValue2) {
+		m_vo.DefaultValue2 = defaultValue2;
+	}
```
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/model/GridField.java:608]: this actually not needed. getVO().DefaultValue and getVO().DefaultValue2 did the same thing.

File: org.adempiere.base/src/org/compiere/model/GridField.java [hunk 2]
```diff
@@ -83,7 +83,15 @@ public class GridField
...
-	private static final long serialVersionUID = 496387784464611123L;
+	private static final long serialVersionUID = 405469916055906825L;
+	
+	private static final Character SPECIAL_CASE_DEFAULT = '1';
+	private static final Character SQL_DEFAULT = '2';
+	private static final Character DEFAULT_LOGIC = '3';
+	private static final Character USER_PREFERENCE_DEFAULT = '4';
+	private static final Character SYSTEM_PREFERENCE_DEFAULT = '5';
+	private static final Character PANEL_PREFERENCE_DEFAULT = '6';
+	private static final Character DATA_TYPE_DEFAULT = '7';
```
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/model/GridField.java:94]: probably should create constant for orderGetDefault in getDefault and defaultSeq in getDefaultForPanel too.

Comment (d-ruiz) [reply org.adempiere.base/src/org/compiere/model/GridField.java:94]: Added in the new commit

File: org.adempiere.base/src/org/compiere/model/GridField.java (hunk 1 above)
Comment (d-ruiz) [reply org.adempiere.base/src/org/compiere/model/GridField.java:612]: You're right, but I think this way is better and more intuitive. Trying to follow the law of Demeter here.

Comment (hengsin) [reply org.adempiere.base/src/org/compiere/model/GridField.java:612]: agree but your ticket is misleading to implied that it is not possible with existing code.

Comment (d-ruiz) [reply org.adempiere.base/src/org/compiere/model/GridField.java:612]: That's true. Just changed the commit message and the ticket description and title to reflect what you mean

**Comment by d-ruiz** [comment]
Hi @hengsin, is there anything else here blocking this ticket that I can address?

**Pull request merged**